charts/kong/templates/deployment.yaml

```diff
@@ -158,21 +158,25 @@ spec:
           protocol: TCP
         {{- end }}
         {{- if (and .Values.proxy.http.enabled .Values.proxy.enabled) }}
+        {{- if .Values.proxy.http.containerEnabled | default true }}
         - name: proxy
           containerPort: {{ .Values.proxy.http.containerPort }}
           {{- if .Values.proxy.http.hostPort }}
           hostPort: {{ .Values.proxy.http.hostPort }}
           {{- end}}
           protocol: TCP
         {{- end }}
+        {{- end }}
         {{- if (and .Values.proxy.tls.enabled .Values.proxy.enabled)}}
+        {{- if .Values.proxy.tls.containerEnabled | default true}}
         - name: proxy-tls
           containerPort: {{ .Values.proxy.tls.containerPort }}
           {{- if .Values.proxy.tls.hostPort }}
           hostPort: {{ .Values.proxy.tls.hostPort }}
           {{- end}}
           protocol: TCP
         {{- end }}
+        {{- end }}
```

Comment:
We generally want all Services and ports to support the same set of configuration, so these need to be repeated across all port definitions.

It'd make sense to add a helper for these so that we can just loop over them and generate like configuration for each (like we do for the Services), but just adding the same clauses to each block is fine if you don't want to add that to scope.

Reply:
I agree, it would make more sense to add a helper. Will make the required change and push. Moreover, will be updating the logic considering your suggestion in this comment https://github.com/Kong/charts/pull/1021#discussion_r1509554696

Reply:
updated the code with special handling of servicePort=0 to handle the use-case.